drivers/serial/CMakeLists.txt

```diff
@@ -31,6 +30,19 @@ zephyr_library_sources_if_kconfig(uart_rv32m1_lpuart.c)
 
 zephyr_library_sources_ifdef(CONFIG_USERSPACE   uart_handlers.c)
 
+if(CONFIG_UART_NS16550)
+  zephyr_library_sources(uart_ns16550.c)
+  foreach(NUM RANGE 0 3)
+    if (CONFIG_UART_NS16550_PORT_${NUM})
+      configure_file(
+        uart_ns16550_port_x.h
```

Comment:
I didn't know this was a cmake thing.  Clever.

Reply:
indeed